### ISSUE TITLE
Use DoubleTensor 

### ISSUE THREAD

**Issue by keyulu** [pyro-ppl/pyro#2560]
Is there a way to use DoubleTensor with Pyro? With variationalsparseGP, I define my kernel and data using torch.DoubleTensor, but I got the following error

`RuntimeRrror: Function AddBackward0 returned an invalid gradient at index 1 - expected type torch.FloatTensor but got torch.DoubleTensor. `

**Comment by fehiepsi** [comment]
@keyulu You can use [torch.set_default_dtype](https://pytorch.org/docs/stable/torch.html#torch.set_default_dtype`) or you need to print out all parameters of your model (a `torch.nn.Module`) to see which one is `float32`.

FYI we have a [forum](https://forum.pyro.ai/) with many Pyro users active. It will be more convenient to ask questions in the forum. I believe you can get more clever solutions to your questions over there. ;)

**Comment by keyulu** [comment]
@fehiepsi  Thank you for the reference, I'll look into the forum ;)
